src/BoundsInference.cpp

```diff
@@ -378,12 +474,13 @@ class BoundsInference : public IRMutator {
 
             // Make the buffer_ts representing the output. They all
             // use the same size, but have differing types.
+            const vector<string> func_args = func.args();
             for (int j = 0; j < func.outputs(); j++) {
                 vector<Expr> output_buffer_t_args(2);
                 output_buffer_t_args[0] = null_handle;
                 output_buffer_t_args[1] = make_zero(func.output_types()[j]);
-                for (size_t k = 0; k < func.args().size(); k++) {
-                    const string &arg = func.args()[k];
+                for (size_t k = 0; k < func_args.size(); k++) {
```

Comment:
Hmm, I'm seeing tons of loops here (both changed and in the old code) that could be range based for loops. Maybe it would be better to do that in a separate change.